### PR TITLE
build(webpack): Change webpack to have smaller chunks and make all entries async 

### PR DESCRIPTION
**Pull request by billyvg**
This changes our entrypoints to dynamically import the main bundles (i.e. the previous entrypoints). The goal here is to make the entrypoints as minimal as possible as we may need these to be uncached for splitting off frontend deploys.

The other change is removing the vendor chunk that we had, and instead use webpack defaults and let it decide how to chunk our build. This results in the client needing to request more + smaller chunks (instead of fewer + larger chunks). This costs a bit of network overhead, but a trade-off here is that the smaller chunks should have longer cache lives.

I'd like to push this out and evaluate how it affects our web vitals in prod and then make a decision on how we'd like to proceed.

Here's a light benchmarking of the perf changes, I used Chrome's dev tools to throttle a mix of the CPU and network bandwidth:

(tbh `DOMContentLoaded` doesn't seem like a very practical measurement here, but it's what was readily available in CDT)
image

(private spreadsheet: https://docs.google.com/spreadsheets/d/1EN2jD0fqAJCASaAL1hAEBZJScTutmCZgCvDooamB3u4/)

### CODE REVIEW

**Comment by EvanPurkhiser** [comment]
approved for staging

File: static/app/bootstrap/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  // If __initialData is not already set on the window, we are likely running in
+  // pure SPA mode, meaning django is not serving our frontend application and we
+  // need to make an API request to hydrate the bootstrap data to boot the app.
+  if (bootstrapData === undefined) {
+    return await bootWithHydration();
+  }
+
+  return bootApplication(bootstrapData);
+}
```
Comment (billyvg) [inline static/app/bootstrap/index.tsx:35]: I've tried to make sure we don't have import side-effects and instead make sure we explicitly invoke fns.

File: static/app/bootstrap/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// These imports (core-js and regenerator-runtime) are replacements for deprecated `@babel/polyfill`
+import 'core-js/stable';
+import 'regenerator-runtime/runtime';
```
Comment (EvanPurkhiser) [inline static/app/bootstrap/index.tsx:3]: Didn't you just move these to webpack?

**Review by evanpurkhiser — APPROVED** [review]
lgtm

File: static/app/bootstrap/index.tsx (hunk 2 above)
Comment (billyvg) [reply static/app/bootstrap/index.tsx:3]: Yeah, those changes didn't land yet

**Pull request merged**